src/decoder/mod.rs

```diff
@@ -1045,6 +1110,14 @@ impl<R: Read + Seek> Decoder<R> {
 
     /// Decodes the entire image and return it as a Vector
     pub fn read_image(&mut self) -> TiffResult<DecodingResult> {
+
+        if self.compression_method == CompressionMethod::ModernJPEG {
+            match self.find_tag(Tag::TileOffsets) {
+                Ok(None) | Err(_) => return self.read_jpeg(true),
```

Comment:
We could argue that ignoring some errors is fine for falling back but _all_ errors including `Io` is too general.